### PR TITLE
Added paritally converted documentation

### PR DESCRIPTION
**Pull request by GeeH**
@weierophinney can you help me out by making a list of mistakes you see here, my idea is currently to open one issue for each step on each repository, even if we don't see that problem, so, for example, the issue may say

### Check the markdown conversion of code examples has done completely in all files

This may result in a comment saying "no problems found" so we can close the issues, I'll write a script that opens the issues automatically then against the right repo.

Thanks!

### CODE REVIEW

File: doc/book/zend.cache.pattern.callback-cache.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+--------
+
+The callback cache pattern caches calls of non specific functions and methods given as a callback.
+
+Quick Start
+-----------
+
+For instantiation you can use the `PatternFactory` or do it manual:
+
+``` sourceCode
```
Comment (weierophinney) [inline doc/book/zend.cache.pattern.callback-cache.md]: Check for gated blocks with:

```markdown
    ``` sourceCode
```

And fix them to reference the appropriate language.

File: doc/book/zend.cache.pattern.callback-cache.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+Zend\\Cache\\Pattern\\CallbackCache
+===================================
```
Comment (weierophinney) [inline doc/book/zend.cache.pattern.callback-cache.md]: Update top-level headers to use the `#` syntax:

```markdown
# Vendor\Component\Classname
```

File: doc/book/zend.cache.pattern.callback-cache.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+Overview
```
Comment (weierophinney) [inline doc/book/zend.cache.pattern.callback-cache.md]: Update second (and third, and fourth, etc.)-level headers to use the `##` notation:

```markdown
    ## Second level heading
    ### Third level heading
    #### Fourth level heading
```

File: doc/book/zend.cache.pattern.callback-cache.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+$callbackCache->setOptions(new PatternOptions(array(
+    'storage'      => 'apc',
+    'cache_output' => true,
+)));
+```
+
+Configuration Options
+---------------------
+
+<table>
```
Comment (weierophinney) [inline doc/book/zend.cache.pattern.callback-cache.md]: Refactor tables using GitHub table syntax (seehttps://help.github.com/articles/github-flavored-markdown/#tables)

File: doc/book/zend.cache.pattern.callback-cache.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+</tbody>
+</table>
+
+Available Methods
+-----------------
+
+Examples
+--------
+
+**Instantiating the callback cache pattern**
```
Comment (weierophinney) [inline doc/book/zend.cache.pattern.callback-cache.md]: Change pseudo-headers that use bold to actual headers.

File: doc/book/zend.cache.pattern.class-cache.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+<td align="left"><code>[]</code></td>
+<td align="left">List of methods to no-cache (If <code>cache_by_default</code> is enabled)</td>
+</tr>
+</tbody>
+</table>
+
+Available Methods
+-----------------
+
+> rtype
```
Comment (weierophinney) [inline doc/book/zend.cache.pattern.class-cache.md]: WTF happened here, @GeeH ?

File: doc/book/zend.cache.pattern.object-cache.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+<td align="left"><code>false</code></td>
+<td align="left">Cache calls of magic object properties</td>
+</tr>
+</tbody>
+</table>
+
+Available Methods
+-----------------
+
+> rtype
```
Comment (weierophinney) [inline doc/book/zend.cache.pattern.object-cache.md]: I'm seeing a pattern; were these definition lists, by any chance?

File: doc/book/zend.cache.storage.adapter.md [hunk 8]
```diff
@@ -0,0 +1,407 @@
+Zend\\Cache\\Storage\\Adapter
+=============================
+
+Overview
+--------
+
+> Storage adapters are wrappers for real storage resources such as memory and the filesystem, using
```
Comment (weierophinney) [inline doc/book/zend.cache.storage.adapter.md]: Flow "blockquote"-style paragraphs using line lengths as documented in the "documentation" repository, starting each line with the `> ` characters, and one `> ` empty line between paragraphs.

File: doc/book/zend.cache.storage.adapter.md [hunk 9]
```diff
@@ -0,0 +1,407 @@
...
+`Zend\Cache\Storage\Adapter\AbstractAdapter`, which comes with basic logic.
+Configuration is handled by either `Zend\Cache\Storage\Adapter\AdapterOptions`, or an
+adapter-specific options class if it exists. You may pass the options instance to the class at
+instantiation or via the `setOptions()` method, or alternately pass an associative array of options
+in either place (internally, these are then passed to an options class instance). Alternately, you
+can pass either the options instance or associative array to the
+`Zend\Cache\StorageFactory::factory` method.
+
+> ## Note
+#### Many methods throw exceptions
```
Comment (weierophinney) [inline doc/book/zend.cache.storage.adapter.md]: Fix headers for notes.

File: doc/book/zend.cache.storage.adapter.md (hunk 8 above)
Comment (weierophinney) [inline doc/book/zend.cache.storage.adapter.md]: BTW, looks like this one is incorrectly marked as a blockquote; likely there was an extra space of indentation on at least the first line.

File: doc/book/zend.cache.storage.adapter.md [hunk 10]
```diff
@@ -0,0 +1,407 @@
...
+Because many caching operations throw an exception on error, you need to catch them manually or you
+can use the plug-in `Zend\Cache\Storage\Plugin\ExceptionHandler` with `throw_exceptions` set to
+`false` to automatically catch them. You can also define an `exception_callback` to log exceptions.
+
+Quick Start
+-----------
+
+> Caching adapters can either be created from the provided `Zend\Cache\StorageFactory` factory, or
```
Comment (weierophinney) [inline doc/book/zend.cache.storage.adapter.md]: Check for incorrectly marked blockquote areas.

File: doc/book/zend.cache.storage.adapter.md [hunk 11]
```diff
@@ -0,0 +1,407 @@
...
+
+The `Zend\Cache\Storage\TaggableInterface` implements methods to mark items with one or more tags
+and to clean items matching tags.
+
+The Apc Adapter
+---------------
+
+> The `Zend\Cache\Storage\Adapter\Apc` adapter stores cache items in shared memory through the
+required PHP extension [APC](http://pecl.php.net/package/APC) (Alternative PHP Cache).
+This adapter implements the following interfaces:
```
Comment (weierophinney) [inline doc/book/zend.cache.storage.adapter.md]: Ensure that lists are separated from content either above or below by an empty line.

File: doc/bookdown.json [hunk 12]
```diff
@@ -0,0 +1,15 @@
+{
+    "title": "Zend\\Cache",
+    "target": "html\/",
+    "content": [
+        "book\/zend.cache.storage.adapter.md",
```
Comment (weierophinney) [inline doc/bookdown.json]: Ensure the `JSON_UNESCAPED_SLASHES` flag is passed to `json_encode()` when generating `bookdown.json` files.

Comment (GeeH) [inline doc/bookdown.json]: OK, can fix in script

**Comment by GeeH** [comment]
I think some of these I've fixed before, let me look at what can be automated, what can't will be opened as tickets once the PR is merged.

**Comment by GeeH** [comment]
Ready to merge, I also created issues on this repo to check you are happy with them @weierophinney

**Pull request merged**